packages/block-library/src/heading/theme.scss

```diff
@@ -0,0 +1,66 @@
+// These follow a Major Third type scale
+h1 {
```

Comment:
Question 1: Should these styles be in `theme.scss`, or `editor.scss`? Currently, `theme.scss` is currently loaded in the editor regardless of whether the theme has opted in, but it has the added benefit of not being output in the theme itself unless opted into.

It seems like we should either load theme.scss only into the editor if opted into, or reconsider what it means for the editor to have a "vanilla stylesheet". See later comments on that.

Reply:
So in the case of the heading Block `theme.scss` exists to

1. Apply default heading styling rules within the Editor
2. Optionally (when opted into) apply heading styling in the Theme when you load front of the website.

Is that right?

Reply:
I'm still unsure about outputting un-scoped element styles in a theme's front-end. Even with opt-in theme.css.

The pros:
- specificity couldn't be any less
- If the theme has this styled at all it's styles will be used
- 98% of themes will have `h*` font-sizes defined.

The questions:
- What if the theme has made a choice to go with the browser defaults for heading font-size or margins.

Essentially what we're doing here and with any naked element styles is creating our own "browser defaults". 
Do we need to add our own styles here at all? The browser defaults aren't that bad.

Reply:
Great thoughts, thank you.

> What if the theme has made a choice to go with the browser defaults for heading font-size or margins. Would they be happy with our modified defaults?
> Essentially what we're doing here and with any naked element styles is creating our own "browser defaults".
> Do we need to add our own styles here at all? The browser defaults aren't that bad.

I agree, the browser defaults aren't bad!

However this is another one of those situations where if we don't provide any styles for the heading sizes, then the _WordPress_ heading sizes will be used because we inherit them. Here's an example:

![Screenshot 2019-03-15 at 11 07 31](https://user-images.githubusercontent.com/1204802/54424144-910c5a00-4712-11e9-9dd0-4834b676b5f7.png)

That's an unscoped `h1` style that the wp-admin stylesheet provides. For h1 it _happens_ to match the browser default of 2em, but this is not hte case with h2 and h3s, and as you can also see it customizes the default font weight of all heading sizes too.

It is frustrating that this bleeds from the wp-admin into the editing canvas, and as mentioned I would like to refactor this away long term by better scoping the admin styles, or by protecting the canvas using Shadow DOM, or any other method we can to prevent or mitigate CSS bleed. 

So this PR is focusing on the most beneficial next step we can take — which is to _still_ normalize those sizes, but at least reduce the specificity so they are easier to override.

My reason for putting these in theme.scss was to provide a theme some means to adopt them without outright copy them. Another option is to put them into `editor.scss`, where they will _ONLY_ be loaded for the editor. But then a theme _has_ to copy them, if they want to build on them. But in this specific case — for headings — maybe this is what we want?

The other perspective is that it can be convenient for a theme to simply opt into styles using 
`add_theme_support( 'wp-block-styles' );`and then the frontend immediately looks like the backend and can then _build from there_. But perhaps this is an argument for a separate opt in, something like `add_theme_support( 'wp-block-normalize' );`, which opts into the same normalization stylesheet that we need for the editing canvas?